### PR TITLE
77 workflow folder name

### PR DESCRIPTION
**Pull request by pchelle**
WIP (I still need to include examples and tests)
I struggled in the rebase of my code due to the many exchanges and modifications brought yesterday, let me know if there are some bits that seem off

The update incorporate the observed data logic we talked about last week
I have included the PlotTask class, which is really simplifying the code in the Workflow run method

### CODE REVIEW

File: R/error-checks.R [hunk 1]
```diff
@@ -120,10 +120,10 @@ validateIsSameLength <- function(...) {
 
 
 validateNoDuplicatedEntries <- function(x) {
-  if ( any(duplicated(x)) ) {
+  if (any(duplicated(x))) {
     logErrorThenStop(messages$errorDuplicatedEntries(deparse(substitute(x))))
   }
-  else{
+  else {
```
Comment (msevestre) [inline R/error-checks.R]: you don't need an else here

File: R/error-checks.R [hunk 2]
```diff
@@ -196,7 +187,13 @@ checkOverwriteExisitingPath <- function(path, overwrite) {
 #'
 #' @return TRUE if the path includes the extension
 isFileExtension <- function(path, extension) {
-  return(grep(pattern = paste0(".", extension), x = path) == 1)
+  return(max(sapply(extension,
```
[the remark this answers is not in the document]
Comment (pchelle) [reply R/error-checks.R]: It is a bit extreme indeed, but this way you can input a vector or list of extensions.
Consequently, if more than one extension is allowed you don't need to loop or anything.
If you have something simpler in mind, I can update

[the remark this answers is not in the document]
Comment (msevestre) [reply R/error-checks.R]: Oh I see
In that case yes, there is definitely easier to read
```
fileExtension <-function(file) {
  ex <- strsplit(basename(file), split="\\.")[[1]]
  return(ex[-1])
}
isFileExtension <- function(file, extension){
  extension <- c(extension)
  file_ext <- fileExtension(file)
  file_ext %in% extension
 }
```

It's more about will you understand what the code does in 6 months.

**Comment by msevestre** [comment]
Going to merge this. I can't review so many changes at once

**Pull request merged**